src/main/java/org/medicmobile/webapp/mobile/Utils.java

```diff
@@ -31,6 +33,27 @@ static boolean isUrlRelated(String appUrl, String uriToTest) {
 		return uriToTest.matches("^(blob:)?" + appUrl + "/.*$");
 	}
 
+	/**
+	 * Valid if navUrl is not null, starts with appUrl as prefix, and
+	 * isn't the login page nor a rewrite path.
+	 */
+	static boolean isValidNavigationUrl(String appUrl, String navUrl) {
+		URL parsedAppUrl;
+		try {
+			parsedAppUrl = new URL(appUrl);
+		} catch (MalformedURLException e) {
+			return false;
+		}
+		String parsedBaseUrl = parsedAppUrl.getProtocol() + "://" + parsedAppUrl.getHost();
+		if (navUrl != null
+				&& navUrl.startsWith(parsedBaseUrl)
+				&& !navUrl.contains("/login")
+				&& !navUrl.contains("/_rewrite")) {
+			return true;
+		}
+		return false;
+	}
+
```

Comment:
The `isUrlRelated` method above is, as far as I can tell, trying to do a similar thing as this function. They're not completely the same because this checks for `null` and `login` and `_rewrite`, but the regex/parse bit is the same. Find a way to reuse the code so they behave consistently.

Reply:
I made the refactor to reuse the code, and moved the null checks to that method to make it safer. Could you please take a look again?